lib/python/qmk/cli/new/keyboard.py

```diff
@@ -1,11 +1,133 @@
-"""This script automates the creation of keyboards.
+"""This script automates the creation of new keyboard directories using a starter template.
 """
+from datetime import date
+import fileinput
+from pathlib import Path
+import re
+import shutil
+
+from qmk.commands import git_get_username
+import qmk.path
 from milc import cli
+from milc.questions import choice, question
+
+KEYBOARD_TYPES = ['avr', 'ps2avrgb']
+
+
+def keyboard_name(name):
+    """Callable for argparse validation.
+    """
+    if not validate_keyboard_name(name):
+        raise ValueError
+    return name
+
+
+def validate_keyboard_name(name):
+    """Returns True if the given keyboard name contains only lowercase a-z, 0-9 and underscore characters.
+    """
+    regex = re.compile(r'^[a-z0-9/_]+$')
```

Comment:
This would allow keyboard names to start with underscore and slash.
The latter should definitely be caught at input time, but the former is up to debate.
My preference is to require alphanumeric character as the first.

```suggestion
    regex = re.compile(r'^[a-z0-9][a-z0-9/_]+$')
```